src/OrchardCore/OrchardCore.Mvc.Core/RazorPages/PageConventionCollectionExtensions.cs

```diff
@@ -46,12 +46,21 @@ public static class PageConventionCollectionExtensions
                 {
                     var route = selector.AttributeRouteModel;
 
-                    if (route.Template.StartsWith(areaFolder, StringComparison.Ordinal))
+                    if (route.Template.StartsWith(areaFolder, StringComparison.Ordinal) || (route.Template == areaName && folderPath == "/"))
                     {
                         route.SuppressLinkGeneration = true;
 
-                        var cleanSubTemplate = route.Template.Substring(areaFolder.Length).TrimStart('/');
-                        var template = AttributeRouteModel.CombineTemplates(folderRoute, cleanSubTemplate);
+                        string template;
+
+                        if (route.Template == areaName && folderPath == "/")
```

Comment:
The convention of the folderPath is that it starts with a `/`
And that it doesn't end with a `/`
But unless if it is only equal to `/`